### DISCUSSION TITLE
How does the preset feature work?

### DISCUSSION THREAD

**Discussion by awacode21** [histoire-dev/histoire#187] [Q&A] [answered]
i am wondering about how does the preset feature in histoire work? what is the intend of this feature? I was not able to find something about it in the docs.
I recognized that every component gets an "initial state" preset where simply nothing at all is set. And i recognized that while being on the histoire app i can manually add presets, for example for red and green button (stackblitz example from histoire). It seems that the preset get saved in local storage and therefore are 'user based' and need to be set again once local storage gets cleared.

I would like to ask if it is not possible to set presets from code base. At the end isn't it the same as the variants, but accessible from a select? And why is the 'initial State' not showing the prop values that got defined for the variant? in initial State all control fields are empty even if the prop values got defined.
<img width="942" alt="Bildschirmfoto 2022-07-20 um 10 43 43" src="https://user-images.githubusercontent.com/30939926/179940962-533b9197-aa38-432e-8a7b-3a473242601e.png">

<img width="951" alt="Bildschirmfoto 2022-07-20 um 10 43 55" src="https://user-images.githubusercontent.com/30939926/179940995-517b20a7-4ef2-4088-91df-03bb05f34796.png">

**Comment by Akryum** [comment, marked as answer]
> At the end isn't it the same as the variants, but accessible from a select? 

It is aimed at the story reader while Variants are for the story author.

> And why is the 'initial State' not showing the prop values that got defined for the variant? in initial State all control fields are empty even if the prop values got defined.

This sounds like a bug, could you please open a new issue about it? Thx

**Reply by awacode21** [reply]
ok, so when i understand it right.. variants from the author should or could prepare all possible variants of a component. But if maybe not all variants are prepared or the user/reader wants to also check for further variants/combinations and wants to save them somehow, so that he does not need to type them in again the next time he comes back he can use presets to make his life easier working with the tool?

**Reply by Akryum** [reply]
That's right :)

**Reply by awacode21** [reply]
Thanks for clarification.

About the empty controls bug i created a ticket: https://github.com/histoire-dev/histoire/issues/188
